src/helpers/pivot/spreadsheet_pivot/date_spreadsheet_pivot.ts

```diff
@@ -0,0 +1,57 @@
+import { toJsDate, toNumber } from "../../../functions/helpers";
+import { Locale } from "../../../types";
+import { PivotDimension } from "../../../types/pivot";
+import { FieldValue } from "./data_entry_spreadsheet_pivot";
+
+export function createDate(dimension: PivotDimension, value: FieldValue["value"], locale: Locale) {
+  const granularity = dimension.granularity || "month_number";
+  if (!(granularity in MAP_VALUE_DIMENSION_DATE)) {
+    throw new Error(`Unknown date granularity: ${granularity}`);
+  }
+  if (!MAP_VALUE_DIMENSION_DATE[granularity].set.has(value)) {
+    MAP_VALUE_DIMENSION_DATE[granularity].set.add(value);
+    const date = toJsDate(value, locale);
+    let number: FieldValue["value"] = 0;
+    switch (granularity) {
+      case "year_number":
+        number = date.getFullYear();
+        break;
+      case "month_number":
+        number = date.getMonth();
+        break;
+      case "day_of_month":
+        number = date.getDate();
+        break;
+      case "day":
+        number = Math.floor(toNumber(value, locale));
+        break;
+    }
+    MAP_VALUE_DIMENSION_DATE[granularity].values[`${value}`] = number;
+  }
+  return MAP_VALUE_DIMENSION_DATE[granularity].values[`${value}`];
+}
+
+/**
+ * This map is used to cache the different values of a pivot date value
+ */
```

Comment:
could you add an example of its shape in the comments like
{ '2012-05-23' : { 'year_number': 2012, ...} , ... }